dask/dataframe/groupby.py

```diff
@@ -305,11 +305,15 @@ def _var_chunk(df, *index):
 
     n = g[x.columns].count().rename(columns=lambda c: (c, "-count"))
 
-    df[cols] = df[cols] ** 2
+    try:
+        for col in set(cols).difference(set(index)):
+            df[col] = df[col] ** 2
+    except TypeError:
+        df[cols] = df[cols] ** 2
+
     g2 = _groupby_raise_unaligned(df, by=index)
     x2 = g2.sum().rename(columns=lambda c: (c, "-x2"))
 
-    x2.index = x.index
```

Comment:
Why this change?

Reply:
This is the specific failure mode: `df.groupby(by=index)` will produce `n` groupings on `index`. `(df**2).groupby(by=index)` will produce `m<n` groupings, in pandas whenever values in `index` are multiples of `np.iinfo(np.int64).max**0.5`, and in `cudf` whenever the values in `index` are simply larger than it.

Reply:
`len(x2.index) != len(x.index)`, and an exception is raised. Will post a reproducer in pandas and cudf soon.

Reply:
It looks like this happens when you exceed max int64.  Like the following:

```python
In [55]: df.iloc[2]                                                                                                                                                                                                                                          
Out[55]: 
x    8589934592
z             2
y             2
Name: 2, dtype: int64

In [56]: df.iloc[2]**2                                                                                                                                                                                                                                       
Out[56]: 
x    0
z    4
y    4
Name: 2, dtype: int64
```

So the resulting groupyby index will be smaller due to more values becoming 0.  However, this does make the operation numerically incorrect

Reply:
That's true, but the failure in this case is when the `by` key/s square to be larger than their maximum. Those keys don't need to be squared, they're just bins for the non-by keys which are used in the calculation of standard deviation. I think. :)